src/Migrations/HSLMigration.php

```diff
@@ -0,0 +1,646 @@
+    }
+
+    return $node->replace($argument_list, $new_argument_list);
+  }
+
+  // many PHP functions can return false, and their HSL counterparts return null instead
+  // this will replace false with null in binary expressions like === false and !=== false
+  protected function maybeChangeFalseToNull(
+    EditableNode $root,
+    FunctionCallExpression $node,
+  ): EditableNode {
+    $parents = null;
+    $found = false;
+    $stack = $root->findWithParents($it ==> $it === $node);
+    invariant(!C\is_empty($stack), 'did not find node in root');
+    invariant(C\lastx($stack) === $node, 'expected node at top of stack');
+    $stack_count = C\count($stack);
+    $parent = $stack[$stack_count - 2];
+
+    if ($parent instanceof BinaryExpression) {
+      if ($parent->getLeftOperand() === $node) {
+        $check = $parent->getRightOperand();
+      } else {
+        $check = $parent->getLeftOperand();
+      }
+
+
+      if ($check instanceof LiteralExpression) {
+        $expression = $check->getExpression();
+        if ($expression instanceof BooleanLiteralToken) {
+          if (Str\lowercase($expression->getText()) === 'false') {
+            $new = new NullLiteralToken(
+              $expression->getLeading(),
+              $expression->getTrailing(),
+            );
+            $root = $root->replace($expression, $new);
+          }
+        }
+      }
+    }
+    return $root;
+  }
+
+  // find all HH\Lib\* namespace use declarations
+  // returns a tuple containing the declaration nodes, the HSL namespace suffixes used
+  protected function findUseDeclarations(
+    vec<INamespaceUseDeclaration> $declarations,
+  ): (vec<INamespaceUseDeclaration>, vec<string>) {
+
+    $search = vec["HH", "Lib"];
+    $nodes = vec[];
+    $suffixes = vec[];
+    foreach ($declarations as $decl) {
+
+      // we only care about "use namespace" directives
+      if (!($decl->getKind() instanceof NamespaceToken)) {
+        continue;
+      }
+
+      if ($decl instanceof NamespaceGroupUseDeclaration) {
+
+        // group declarations: does prefix match?
+        $parts =
+          $decl->getPrefix()->getParts()->getItemsOfType(NameToken::class);
+        if (C\count($parts) !== 2) {
+          continue;
+        }
+        $found_prefix = true;
+        foreach ($parts as $i => $token) {
+          if ($token->getText() === $search[$i]) {
+            continue;
+          }
+
+          $found_prefix = false;
+          break;
+        }
+
+        // prefix didn't match? skip this declaration
+        if (!$found_prefix) {
+          continue;
+        }
+
+        // prefix matches: add node to list of nodes
+        $nodes[] = $decl;
+
+        $clauses = vec(
+          $decl->getClauses()->getDescendantsOfType(NamespaceUseClause::class),
+        );
+        foreach ($clauses as $clause) {
+          $name = $clause->getName();
+          if ($name instanceof NameToken) {
+            $suffixes[] = $name->getText();
+          }
+        }
```

Comment:
This is fine - but alternatively (up to you):

```Hack
$suffixes = $clauses
  |> Vec\map($$, $c ==> { $n = $c->getName(); return $n instanceof NameToken ? $n->getText(): null; })
  |> Vec\filter_nulls($$)
  |> Vec\concat($suffixes, $$);
```

This is also another case where it feels like `keyset[]` is the right approach

Reply:
I used this suggestion, thanks!